src/components/LHNOptionsList/LHNOptionsList.js

```diff
@@ -27,14 +35,86 @@ const propTypes = {
 
     /** Whether to allow option focus or not */
     shouldDisableFocusOptions: PropTypes.bool,
+
+    /** The policy which the user has access to and which the report could be tied to */
+    policy: PropTypes.shape({
+        /** The ID of the policy */
+        id: PropTypes.string,
+        /** Name of the policy */
+        name: PropTypes.string,
+        /** Avatar of the policy */
+        avatar: PropTypes.string,
+    }),
+
+    /** All reports shared with the user */
+    reports: PropTypes.objectOf(reportPropTypes),
+
+    /** Array of report actions for this report */
+    reportActions: PropTypes.objectOf(PropTypes.shape(reportActionPropTypes)),
+
+    /** Indicates which locale the user currently has selected */
+    preferredLocale: PropTypes.string,
+
+    /** List of users' personal details */
+    personalDetails: PropTypes.objectOf(participantPropTypes),
+
+    /** The transaction from the parent report action */
+    transactions: PropTypes.objectOf(
+        PropTypes.shape({
+            /** The ID of the transaction */
+            transactionID: PropTypes.string,
+        }),
+    ),
+    /** List of draft comments */
+    comments: PropTypes.objectOf(PropTypes.string),
 };
 
 const defaultProps = {
     style: styles.flex1,
     shouldDisableFocusOptions: false,
+    reportActions: {},
+    reports: {},
+    policy: {},
+    preferredLocale: CONST.LOCALES.DEFAULT,
+    personalDetails: {},
+    transactions: {},
+    comments: {},
 };
 
-function LHNOptionsList({style, contentContainerStyles, data, onSelectRow, optionMode, shouldDisableFocusOptions}) {
+const keyExtractor = (item) => item;
+
+function LHNOptionsList({
+    style,
+    contentContainerStyles,
+    data,
+    onSelectRow,
+    optionMode,
+    shouldDisableFocusOptions,
+    reports,
+    reportActions,
+    policy,
+    preferredLocale,
+    personalDetails,
+    transactions,
+    comments,
+}) {
+    const itemPersonalDetails = useMemo(
+        () =>
+            _.reduce(
+                personalDetails,
+                (finalPersonalDetails, personalData, accountID) => {
+                    // It's OK to do param-reassignment in _.reduce() because we absolutely know the starting state of finalPersonalDetails
+                    // eslint-disable-next-line no-param-reassign
+                    finalPersonalDetails[accountID] = {
+                        ...personalData,
+                        avatar: UserUtils.getAvatar(personalData.avatar, personalData.accountID),
```

Comment:
maybe we can skip reducing() in here for just an avatar and get it further own the tree, only when needed? This feels like is really costly.

Reply:
Yeah, I agree with that too 👍

Reply:
after double-checking, I think we might skip this reduce fn entirely, as the avatars are already set correctly

![image](https://github.com/Expensify/App/assets/13985840/2ba4fd7c-72cb-40b1-a885-6599f6013a07)

Reply:
the more you know! 😅 Perfect, thanks for taking the time!!